Datadog.Serverless.Compat/DatadogServerlessCompat.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+
+namespace Datadog.Serverless.Compat
+{
+    public enum CloudEnvironment
+    {
+        AzureFunction,
+        AzureSpringApp,
+        GoogleCloudRunFunction1stGen,
+        Unknown
+    }
+
+    public static class DatadogServerlessCompat
+    {
+        private static readonly string OS = RuntimeInformation.OSDescription.ToLower();
+        private static readonly string? BinaryPath = Environment.GetEnvironmentVariable(
+            "DD_SERVERLESS_COMPAT_PATH"
+        );
+        private static readonly ILogger _logger;
+
+        static DatadogServerlessCompat()
+        {
+            var logLevelEnv = Environment.GetEnvironmentVariable("DD_LOG_LEVEL");
+
+            var logLevelMapping = new Dictionary<string, LogLevel>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "OFF", LogLevel.None },
+                { "CRITICAL", LogLevel.Critical },
+                { "ERROR", LogLevel.Error },
+                { "WARN", LogLevel.Warning },
+                { "INFO", LogLevel.Information },
+                { "DEBUG", LogLevel.Debug },
+                { "TRACE", LogLevel.Trace }
+            };
+
+            LogLevel logLevel = LogLevel.Information;
+            if (
+                !string.IsNullOrEmpty(logLevelEnv)
+                && logLevelMapping.TryGetValue(logLevelEnv, out var mappedLevel)
+            )
+            {
+                logLevel = mappedLevel;
+            }
```

Comment:
It's a bit overkill to allocate and populate a `Dictionary` for a single lookup. We can use a `switch` expression instead:
```suggestion
            var logLevel = logLevelEnv?.ToUpper() switch
            {
                "OFF" => LogLevel.None,
                "CRITICAL" => LogLevel.Critical,
                "ERROR" => LogLevel.Error,
                "WARN" => LogLevel.Warning,
                "INFO" => LogLevel.Information,
                "DEBUG" => LogLevel.Debug,
                "TRACE" => LogLevel.Trace,
                _ => LogLevel.Information
            };
```